ci-operator/step-registry/cucushift/upgrade/eus/unpause-worker-mcp/cucushift-upgrade-eus-unpause-worker-mcp-commands.sh

```diff
@@ -19,32 +19,147 @@ function debug() {
+    echo "Running worker version: "
+    echo $running_worker_version
+
+    if [[ "$paused_worker_version" == "$running_worker_version" ]]; then
+        echo "Paused worker and unpaused worker have same version, it is not corect."
+        exit 1
+    fi
+}
+
+# After all mcp upgraded, all worker should on same version
+function post_check_mcp(){
+    echo "Post check, after all mcp upgraded, all worker should on same version"
+    local paused_worker_version running_worker_version
+    paused_worker_version=$(oc get -l 'node-role.kubernetes.io/'${PAUSED_MCP_LABEL}'=' nodes -ojson | jq '.items[0].status.nodeInfo' | jq -r '.containerRuntimeVersion,.kernelVersion,.osImage')
+    echo "Paused worker version: "
+    echo $paused_worker_version
+    running_worker_version=$(oc get -l 'node-role.kubernetes.io/'${NORMAL_MCP_LABEL}'=' nodes -ojson | jq '.items[0].status.nodeInfo' | jq -r '.containerRuntimeVersion,.kernelVersion,.osImage')
+    echo "Running worker version: "
+    echo $running_worker_version
+
+    if [[ "$paused_worker_version" != "$running_worker_version" ]]; then
+        echo "Paused worker and unpaused worker should have same version but actually not."
+        exit 1
+    fi
+}
+
+# unpause the paused mcp
 function unpause() {
-    oc patch --type=merge --patch='{"spec":{"paused":false}}' machineconfigpool/worker
-    ret=$(oc get machineconfigpools worker -ojson| jq .spec.paused)
+    echo "Unpause the paused mcp..."
+    oc patch --type=merge --patch='{"spec":{"paused":false}}' mcp/${PAUSED_MCP_LABEL}
+    if [ $? -ne 0 ]; then
+        echo "Unpause mcp failed"
+        exit 1
+    fi
+
+    ret=$(oc get machineconfigpools ${PAUSED_MCP_LABEL} -ojson | jq -r '.spec.paused')
     if [[ "${ret}" != "false" ]]; then
-        echo >&2 "Failed to resume worker pool, exiting..." && return 1
+        echo >&2 "${PAUSED_MCP_LABEL} pool failed to unpause, exiting" && exit 1
     fi
 }
 
 function check_mcp() {
     local out updated updating degraded try=0 max_retries=30
+    local label="$1"
+    echo "Check mcp with ${label}..."
     while (( try < max_retries )); 
     do
+        sleep 300
         echo "Checking worker pool status #${try}..."
         run_command "oc get machineconfigpools"
-        out="$(oc get machineconfigpools worker --no-headers)"
+        out="$(oc get machineconfigpools ${label} --no-headers)"
         updated="$(echo "${out}" | awk '{print $3}')"
         updating="$(echo "${out}" | awk '{print $4}')"
         degraded="$(echo "${out}" | awk '{print $5}')"
     
         if [[ ${updated} == "True" && ${updating} == "False" && ${degraded} == "False" ]]; then
             echo "Worker pool status check passed" && return 0
         fi  
-        sleep 300
         (( try += 1 ))
     done  
-    echo >&2 "Worker pool status check failed" && return 1
+    echo >&2 "Worker pool status check failed" && exit 1
+}
+
+function remove_mcp_label(){
```

Comment:
Suggest to develop a new step like `mco-conf-day2-destroy-custom-mcp` corresponding to `mco-conf-day2-create-custom-mcp` to make this step only focus on unpause mcp operations.